parastell/invessel_build.py

```diff
@@ -1265,14 +1269,45 @@ def gap_map(self, list):
 
         self._gap_map = list
 
+    def _add_tags_to_core(self):
+        """Creates PyMOAB core instance with source strength tag.
+        (Internal function not intended to be called externally)
+        """
+        tag_type = types.MB_TYPE_DOUBLE
+        tag_size = 1
+        storage_type = types.MB_TAG_DENSE
+
+        vol_tag_name = "Volume"
+        self.volume_tag = self.mbc.tag_get_handle(
+            vol_tag_name,
+            tag_size,
+            tag_type,
+            storage_type,
+            create_if_missing=True,
+        )
+
     def create_vertices(self):
         """Creates mesh vertices and adds them to PyMOAB core."""
-        coords = []
+        self.coords = []
         for surface in self.surfaces:
             for rib in surface.Ribs:
-                coords.extend(rib.rib_loci)
-        coords = np.array(coords)
-        self.add_vertices(coords)
+                self.coords.extend(rib.rib_loci)
+        self.coords = np.array(self.coords)
+        self.add_vertices(self.coords)
+
+    def _compute_tet_data(self, tet_ids, tet):
+        """Computes tetrahedron volume, and sets the corresponding value of
+        the respective tag for that tetrahedron.
+        (Internal function not intended to be called externally)
+
+        Arguments:
+            tet_ids (list of int): tetrahedron vertex indices.
+            tet (object): pymoab.EntityHandle of tetrahedron.
+        """
+        tet_vol = self._compute_tet_volume(tet_ids)
+        self.volumes.append(tet_vol)
```

Comment:
Seems a little buried to update this list here.  Could you return the volume and then append it elsewhere?